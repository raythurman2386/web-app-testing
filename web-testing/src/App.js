import React from 'react'

function App() {
  return (
    <section aria-labelledby='KittensHeader'>
      <h2 id='KittensHeader'>All About Kittens</h2>
      <p className='content'>Lorem ipsum dolor sit amet</p>
    </section>
  )
}

export default App
