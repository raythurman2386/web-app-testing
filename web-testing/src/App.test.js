import React from 'react'
import * as rtl from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import renderer from 'react-test-renderer'
import App from './App'
import Dashboard from './components/Dashboard'
import Display from './components/Display'

afterEach(rtl.cleanup)

describe('My Test Suite', () => {
  it('Searches for at bat text', () => {
    const wrapper = rtl.render(<App />)

    const atBatText = wrapper.getByText(/at bat/i)
    // assertions will be here
    expect(atBatText).toBeInTheDocument()
  })

  it('Checks how many buttons there are', () => {
    const wrapper = rtl.render(<Dashboard />)
    const checkButtons = wrapper.getAllByRole('button')

    // assertion
    expect(checkButtons.length).toBe(4)
  })

  it('should snapshot ball renders', () => {
    const component = renderer.create(<Display ball={0} />)
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should snapshot strike renders', () => {
    const component = renderer.create(<Display strike={0} />)
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
