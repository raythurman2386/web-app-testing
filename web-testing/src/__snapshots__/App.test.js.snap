// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My Test Suite should snapshot ball renders 1`] = `
<div>
  <h1>
    At Bat Stats
  </h1>
  <p>
    Balls: 
    0
  </p>
  <p>
    Strikes: 
  </p>
</div>
`;

exports[`My Test Suite should snapshot renders 1`] = `
<div>
  <h1>
    At Bat Stats
  </h1>
  <p>
    Balls: 
    0
  </p>
  <p>
    Strikes: 
  </p>
</div>
`;

exports[`My Test Suite should snapshot strike renders 1`] = `
<div>
  <h1>
    At Bat Stats
  </h1>
  <p>
    Balls: 
  </p>
  <p>
    Strikes: 
    0
  </p>
</div>
`;
